File: 프로그래머스/1/340198. ［PCCE 기출문제］ 10번 ／ 공원/［PCCE 기출문제］ 10번 ／ 공원.py
def solution(mats, park):
    mats.sort(reverse=True)
    
    N = len(park)
    M = len(park[0])
    
    for size in mats:
        for i in range(N - size + 1):
            for j in range(M - size + 1):
                flag = True
                
                for x in range(i, i+size):
                    for y in range(j, j+size):
                        if park[x][y] != "-1":
                            flag = False
                            break
                    if not flag:
                        break
                        
                if flag:
                    return size
    return -1

File: 프로그래머스/1/340198. ［PCCE 기출문제］ 10번 ／ 공원/［PCCE 기출문제］ 10번 ／ 공원.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> mats, vector<vector<string>> park) {
    sort(mats.rbegin(), mats.rend());
    
    int row = park.size();
    int col = park[0].size();
    
    for (int size : mats) {
        for (int i = 0; i + size <= row; i++) {
            for (int j = 0; j + size <= col; j++) {
                bool possible = true;
                
                for (int r = i; r < i + size; r++) {
                    for (int c = j; c < j + size; c++) {
                        
                        if (park[r][c] != "-1") {
                            possible = false;
                            break;
                        }
                    }
                    
                    if (!possible) {
                        break;
                    }
                }

                if (possible) {
                    return size;
                }
            }
        }
    }
    
    return -1;
}